File: src/main/java/me/anviks/codewars/_6kyu/build_tower/SolutionTest.java
/*
 * https://www.codewars.com/kata/576757b1df89ecf5bd00073b
 */

package me.anviks.codewars._6kyu.build_tower;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SolutionTest {
    @Test
    public void basicTests() {
        assertEquals(String.join(",", "*"), String.join(",", Kata.towerBuilder(1)));
        assertEquals(String.join(",", " * ", "***"), String.join(",", Kata.towerBuilder(2)));
        assertEquals(String.join(",", "  *  ", " *** ", "*****"), String.join(",", Kata.towerBuilder(3)));
    }
}

File: src/main/kotlin/me/anviks/codewars/translations/_6kyu/split_strings/SolutionTest.kt
package me.anviks.codewars.translations._6kyu.split_strings

import kotlin.random.Random
import kotlin.test.Test
import kotlin.test.assertEquals


class SolutionTest {
    @Test
    fun basicTests() {
        assertEquals(listOf(), solution(""), "Should handle empty string")
        assertEquals(listOf("ab", "cd", "ef"), solution("abcdef"), "Should handle even string")
        assertEquals(listOf("ab", "cd", "ef", "g_"), solution("abcdefg"), "Should handle odd string")
    }

    @Test
    fun spacesTests() {
        assertEquals(listOf("H ", " e", "l ", " l", "o_"), solution("H  el  lo"), "Should handle spaces")
        assertEquals(listOf("w ", "  ", "o ", " r", "l ", "d_"), solution("w   o  rl d"), "Should handle spaces")
    }

    @Test
    fun randomTests() {
        val random = Random.Default
        val chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ       1234567890______"

        repeat(300) {
            val parts = mutableListOf<String>()
            repeat(random.nextInt(1, 10)) {
                parts.add(chars[random.nextInt(chars.length)].toString())
            }
            val str = parts.joinToString("")
            val expected = str.chunked(2).map { it.padEnd(2, '_') }
            assertEquals(expected, solution(str), "Failed with input: $str")
        }
    }
}